tests/SprykerEcoTest/Zed/Easycredit/AbstractEasycreditTest.php

```diff
@@ -0,0 +1,328 @@
+use SprykerEco\Zed\Easycredit\Persistence\EasycreditEntityManager;
+use SprykerEco\Zed\Easycredit\Persistence\EasycreditEntityManagerInterface;
+use SprykerEco\Zed\Easycredit\Persistence\EasycreditRepositoryInterface;
+
+/**
+ * @group SprykerEcoTest
+ * @group Zed
+ * @group Easycredit
+ * @group EasycreditTest
+ */
+abstract class AbstractEasycreditTest extends Unit
+{
+    protected const RESPONSE_KEY_PAYMENT_IDENTIFIER = 'payment_identifier';
+    protected const RESPONSE_KEY_STATUS = 'status';
+    protected const RESPONSE_KEY_TEXT = 'text';
+    protected const RESPONSE_KEY_ANFALLENDE_ZINSEN = '123.45';
+    protected const RESPONSE_KEY_URL_VORVERTRAGLICHE_INFORMATIONEN = 'url';
+    protected const RESPONSE_KEY_TILGUNGSPLAN_TEXT = 'text';
+
+    /**
+     * @var EasycreditTester
+     */
+    protected $tester;
+
+    /**
+     * @return \SprykerEco\Zed\Easycredit\Business\EasycreditFacade
+     */
+    protected function prepareFacade(): EasycreditFacade
+    {
+        $facade = new EasycreditFacade();
+        $facade->setFactory($this->createEasycreditBusinessFactoryMock());
+
+        return $facade;
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Easycredit\Business\EasycreditBusinessFactory
+     */
+    protected function createEasycreditBusinessFactoryMock(): EasycreditBusinessFactory
+    {
+        $factory = $this->getMockBuilder(EasycreditBusinessFactory::class)
+            ->setMethods([
+                'getEntityManager',
+                'getConfig',
+                'createRequestSender',
+                'createResponseParser',
+                'createAdapterFactory',
+                'createMapper',
+            ])
+            ->getMock();
+
+        $factory->method('getEntityManager')->willReturn(new EasycreditEntityManager());
+        $factory->method('getConfig')->willReturn($this->getConfigMock());
+        $factory->method('createRequestSender')->willReturn($this->getRequestSender());
+
+        return $factory;
+    }
+
+    /**
+     * @return \Generated\Shared\Transfer\QuoteTransfer
+     */
+    protected function prepareQuoteTransfer(): QuoteTransfer
+    {
+        $quoteTransfer = new QuoteTransfer();
+
+        return $quoteTransfer;
+    }
+
+    /**
+     * @return \Generated\Shared\Transfer\PaymentMethodsTransfer
+     */
+    protected function preparePaymentMethodsTransfer(): PaymentMethodsTransfer
+    {
+        $paymentMethodTransfer = new PaymentMethodTransfer();
+        $paymentMethodTransfer->setMethodName(EasycreditConfig::PAYMENT_METHOD);
+
+        $paymentMethodsTransfer = new PaymentMethodsTransfer();
+        $paymentMethodsTransfer->addMethod($paymentMethodTransfer);
+
+        return $paymentMethodsTransfer;
+    }
+
+    /**
+     * @return \Generated\Shared\Transfer\PaymentTransfer
+     */
+    protected function preparePaymentTransfer(): PaymentTransfer
+    {
+        $paymentTransfer = new PaymentTransfer();
```

Comment:
Not a blocker, but you might want to use databuilder to generate test transfers

Reply:
I don't have time for rewriting. We discussed this way with Roman Shopin and agreed that is not really needed there.